### PR TITLE
SDIT-877 Creation of hearing results

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        assertThat(hearingId).isNotNull
+
+        // confirm hearing has been updated with an adjudicator
+        webTestClient.get().uri("/adjudications/hearings/${existingHearing.id}")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ADJUDICATIONS")))
+          .exchange()
+          .expectStatus().isOk
+          .expectBody()
+          .jsonPath("hearingStaff.staffId").isEqualTo(reportingStaff.id)
+          .jsonPath("hearingStaff.username").isEqualTo("JANESTAFF")
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt:199]: more tests to come once GET endpoint added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt [hunk 2]
```diff
@@ -569,6 +569,75 @@ class AdjudicationResource(
...
+      ApiResponse(
+        responseCode = "404",
+        description = "Hearing does not exist",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationResource.kt:628]: Didnt know you could have 2 separate ApiResponses with the same status !

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  private var aLocationInMoorland = -41L
+  private var aSecondLocationInMoorland = -42L
+
+  @Autowired
+  private lateinit var nomisDataBuilder: NomisDataBuilder
+
+  @BeforeEach
+  fun setUp() {
+    aLocationInMoorland = repository.getInternalLocationByDescription("MDI-1-1-001", "MDI").locationId
+    aSecondLocationInMoorland = repository.getInternalLocationByDescription("MDI-1-1-002", "MDI").locationId
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt:35]: Does this set the location ids to be different from the initial values of -41 and -42?

**Review by steverendell — APPROVED** [review]
lgtm with minor comments

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt (hunk 3 above)
Comment (lbennett-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt:35]: looks like duplication thanks

**Pull request merged**